### DISCUSSION TITLE
Deadlocks when running with more than 1 process

### DISCUSSION THREAD

**Discussion by victorsilent** [paratestphp/paratest#819] [Q&A]
There's a way to improve or handle this? I'm using a mysql database and running some tests I got deadlocks, if I try to run it with just one thread it pass fine.

Tried with -f, runners and etc but nothing helped me =/

**Comment by Slamdunk** [comment]
Hi, I run heavily parallelized tests each with a database connection. Do you split each process on a dedicated mysql database?

**Comment by victorsilent** [comment]
I have just a single database, how do you do it?

**Comment by Slamdunk** [comment]
It is pointed out in the readme: https://github.com/paratestphp/paratest#test-token

Also take a loot at: https://github.com/paratestphp/paratest/issues/480#issuecomment-691823517

**Comment by victorsilent** [comment]
Thanks! I checked both things and I have these questions:
Since I know how much process I will run, can I just create the databases on my own (db_1, db_2...)? Also, how can I tell to Laravel what should be the right database name to pick? I see the code ton test-token doc, but not sure how should I use it to achieve the desired result :(

**Comment by Slamdunk** [comment]
> Since I know how much process I will run, can I just create the databases on my own (db_1, db_2...)?

Yes, but test suite should be self-contained to avoid false-positives and false-negatives, so it should be the test suite that creates the database, not you manually

> Also, how can I tell to Laravel what should be the right database name to pick?

That's beyond the scope of this library and my knowledge, sorry :\

**Comment by victorsilent** [comment]
I see, I will start a search and try to figure out the Laravel part, really thanks!

**Comment by victorsilent** [comment]
Hello again! I did it but now I'm getting "PDOException: SQLSTATE[42000]: Syntax error or access violation: 1305 SAVEPOINT trans2 does not exist"
have have you ever seen this error before? Also how do you handle multiple connections on different databases? I'm not sure but I guess this is my issue right now.

**Comment by victorsilent** [comment]
Sorry to ask again haha.
Solved the connections issue, I was changing the connection, not the database name. Now changing just the database name it works right, but even doing this I'm still getting dead locks =/
